apps/aecore/src/aec_chain_state.erl

```diff
@@ -341,17 +346,16 @@ persist_state(OldState, NewState) ->
             true
     end.
 
-db_write_top_block_node(#node{ header = Header
-                             , hash   = Hash }) ->
+-spec db_write_top_block_node(chain_node()) -> ok | no_return().
+db_write_top_block_node(N) ->
```

Comment:
Consider moving this function to aec_chain_node module

Reply:
I think I've addressed most of the issues except this one. Please elaborate, why would we want to move any DB functionality into a data type module? I may be wrong, but is there any perf improvements to be achieved here, and if any, would it be worth loosing the abstraction purity?